Ws  015

Aude11 opened the pull request:
added meta descriptions, authors and keywords

Aude11 committed:
WS--011 added footer to layout html file and added style to the footer with class container footer

Aude11 committed:
added in css file width to 100% for the footer

Aude11 committed:
added flex boxes

Aude11 committed:
made footer responsive via flex

Aude11 committed:
added link to  icone github

Aude11 committed:
added js to update date

Aude11 committed:
added meta descriptions, keywords and authors

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -1,3 +1,12 @@
+Medium-Blue: #064172
+Darker-Blue: #1D77C3
+Lightest-Blue: #42BFDD
+*/
```
There's a really cool feature in css, called css custom properties. Instead of keeping the colours in a comment and copy/pasting, we can keep one source of truth, and reuse variables throughout the css.

```suggestion
:root {
  --darkest-blue: #062844;
  --medium-blue: #062844;
  --darker-blue: #1d77c3;
  --lightest-blue: #42bfdd;
}
```

and then we can use the varables in other elements like so:
```
body {
  background: var(--lightest-blue)
}

```

Alternatively, we can add a bit of semantics and remove the association with the colour itself like so (ignore the colours)
```
:root {
  --accent: #062844;
  --accent-darker: #062844;
  --accent-dark: #1d77c3;
  --accent-light: #42bfdd;
  
  --background-light: #230123;
  --background: #ab32ff;
  
  --primary-color: #111111;
}
```

I do this a lot with my personal web projects, e.g. https://github.com/AndrewYHuang/love-jar/blob/main/src/style.css

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -1,3 +1,12 @@
+Darker-Blue: #1D77C3
+Lightest-Blue: #42BFDD
+*/
+
+
```
Just a couple of unused blank lines here

AndrewYHuang commented on src/main/resources/static/css/application.css:
```diff
@@ -50,3 +59,47 @@
+  font-weight: 500;
+}
+
+
+
+
+
+
+
```
Also a few blank newlines here

Aude11 replied:
Ta

AndrewYHuang commented on src/main/resources/templates/layout.html:
```diff
@@ -40,5 +44,22 @@
     </div>
+
+
+    <footer>
+        <div class="footer-container">
+            <div>WhaleSpotting</div>
+            <div><i class="fa fa-github"></i><a  href="https://github.com/charlottemothersole/WhaleSpotting-Java"> Github repository </a></div>
+            <div>©Copyright <span id="currentYear"></span>, BootCamp Project Amex</div>
+            <script>
+                    console.log("here");
+                    const currentDate = new Date();
+                    let currentYear = currentDate.getFullYear();
+                    document.getElementById("currentYear").innerHTML = currentYear;
+            </script>
+        </div>
+
+    </footer>
```
You might not need the extra div in here, you _might_ just be able to do the below.

Additionally, I'd call what you have in the footer more like `span`s than `div`s
```suggestion
    <footer class="footer-container">
        <span>WhaleSpotting</span>
        <span><i class="fa fa-github"></i><a  href="https://github.com/charlottemothersole/WhaleSpotting-Java"> Github repository </a></span>
        <span>©Copyright <span id="currentYear"></span>, BootCamp Project Amex</span>
        <script>
                const currentDate = new Date();
                let currentYear = currentDate.getFullYear();
                document.getElementById("currentYear").innerHTML = currentYear;
        </script>
    </footer>
```

Aude11 replied:
The divs help with the flex box to allow the footer being responsive.

AndrewYHuang reviewed (changes requested):
Looks good so far :) Just a few suggestions and thoughts

Aude11 committed:
Update src/main/resources/static/css/application.css

Add feature css custom properties as suggested by Andy that allows to keep one source of true

Co-authored-by: Andrew Huang <44811850+AndrewYHuang@users.noreply.github.com>

AndrewYHuang reviewed (approved):
Looks good!

Aude11 committed:
Merge branch 'main' into WS--015
